pkgs/applications/editors/emacs/elisp-packages/melpa-packages.nix

```diff
@@ -484,29 +484,12 @@ let
 
         rime = super.rime.overrideAttrs (old: {
           buildInputs = (old.buildInputs or [ ]) ++ [ pkgs.librime ];
-          preBuild = (old.preBuild or "") +
-          (if pkgs.stdenv.isDarwin then
-            ''
-              export MODULE_FILE_SUFFIX=".dylib"
-              make lib
-              mkdir -p /tmp/build/rime-lib
-              cp *.dylib /tmp/build/rime-lib
-            ''
-          else
-          ''
-            make lib
-            mkdir -p /build/rime-lib
-            cp *.so /build/rime-lib
-          '');
-          postInstall = (old.postInstall or "") +
-          (if pkgs.stdenv.isDarwin then
-          ''
-            install -m444 -t $out/share/emacs/site-lisp/elpa/rime-* /tmp/build/rime-lib/*.dylib
-          ''
-          else
-          ''
-            install -m444 -t $out/share/emacs/site-lisp/elpa/rime-* /build/rime-lib/*.so
-          '');
+          preBuild = (old.preBuild or "") + ''
+            make lib CC=$CC MODULE_FILE_SUFFIX=${pkgs.stdenv.hostPlatform.extensions.sharedLibrary}
```

Comment:
FYI, there is `pkgs.stdenv.hostPlatform.extensions.sharedLibrary`.

https://github.com/NixOS/nixpkgs/pull/322873 @JohnRTitor @klchen0112